lib/python/bind_data_array.h

```diff
@@ -64,6 +64,65 @@ void bind_common_mutable_view_operators(pybind11::class_<T, Ignored...> &view) {
       });
 }
 
+template <class D> auto cast_to_dict_key(const py::handle &obj) {
+  using key_type = typename D::key_type;
+  if constexpr (std::is_same_v<key_type, std::string>) {
+    return obj.cast<std::string>();
+  } else {
+    return key_type{obj.cast<std::string>()};
+  }
+}
+
+template <class D> auto cast_to_dict_value(const py::handle &obj) {
+  using val_type = typename D::mapped_type;
+  return obj.cast<val_type>();
+}
+
+template <class T, class... Ignored, class Set>
+void bind_dict_update(pybind11::class_<T, Ignored...> &view, Set &&set_item) {
+  view.def(
+      "update",
+      [set_item](T &self, const py::object &other, const py::kwargs &kwargs) {
```

Comment:
It seems a fair amount of code could be reused for #2491?

Reply:
Not sure. That issue would only require the first loop. Unless you also want to support passing an iterable of tuples.

Reply:
It disappeared now...